CMakeLists.txt

```diff
@@ -83,6 +83,7 @@ set(TOOLCHAIN_PATH "" CACHE STRING "Path to toolchain directory")
 
 add_compile_definitions(FSL_RTOS_FREE_RTOS
     PLATFORM_${PROJECT_TARGET_NAME}
+    PRODUCT_${PRODUCT}
```

Comment:
My reasoning was that since we already have global `PLATFORM_XXX` then we might also need `PRODUCT_XXX` esspecially considering that AFAIU something like this was also proposed by Grinn: https://github.com/mudita/MuditaOS/pull/4410/files

Reply:
IMHO, common parts of codebase(used by various products) should not contain product-specific logic. I don't think the solution proposed by Grinn is a good one. That's why we have `module-bsp/board/rt1051/<SPECIFIC_PRODUCT>` directories. For a reference, you can check the directory layout of popular rtoses like Nuttx, ZephyrOS or even linux kernel. They all follow basically the same rule(more or less).

Reply:
I removed it and defined product-specific `TAGSFETCHER_MAX_TAG_SIZE` which is then used in tagsfetcher.